doc/source/whatsnew/v0.24.0.txt

```diff
@@ -343,6 +343,8 @@ Other
 ^^^^^
 
 - :meth: `~pandas.io.formats.style.Styler.background_gradient` now takes a ``text_color_threshold`` parameter to automatically lighten the text color based on the luminance of the background color. This improves readability with dark background colors without the need to limit the background colormap range. (:issue:`21258`)
+- Building pandas for development now requires ``cython >= 0.28``
+- Require at least 0.28 version of cython to support read-only memoryviews (:issue:`21688`)
```

Comment:
so we really need to bump cython here?

Reply:
Assuming we need to bump Cython, it looks like there are a few things missing in order to fully bump it: pin some CI builds to the minimum Cython version, update version number in `install.rst`.  

See the changes in #18623 for where I bumped to 0.24.0; should show all the places that need updating.

Reply:
Reading through the changelog of Cython, we need to bump to 0.28 to support the `const` specifier for memoryviews. Looking a bit through the Travis setup, we only pin Cython for the 2.7 builds and these are the ones that failed. (sadly with a weird error message).

Reply:
> we need to bump to 0.28 to support the const specifier for memoryviews

is this an actual change in cython? 

I agree its nice to change this, but cython 0.28 is pretty new.

Reply:
`0.28.0` works but we should build with `0.28.2` due to the mentioned problems in `sklearn`. I would at least bump the Cython version on Travis & CricleCI. 

Should we then allow `0.28.0` or require `0.28.2` so that we don't get the bug reports w.r.t. problems that are fixes in `.2`?